lib/HLSL/HLOperations.cpp

```diff
@@ -509,9 +509,15 @@ Function *GetOrCreateHLFunction(Module &M, FunctionType *funcTy,
   }
   else {
     mangledNameStr << GetHLFullName(group, opcode);
-    // Need to add wave sensitivity to name to prevent clashes with non-wave intrinsic
-    if(attribs.hasAttribute(AttributeSet::FunctionIndex, HLWaveSensitive))
-        mangledNameStr << "wave";
+    // Need to add attributes to name to prevent clashes with intrinsics with
+    // different attributes
+    if (attribs.hasAttribute(AttributeSet::FunctionIndex, Attribute::ReadNone))
+      mangledNameStr << "rn";
```

Comment:
I'm surprised there are no test updates with `rn` - do we not have any ops that have this attribute or do we just not have any tests for them? Similarly, it'd be nice to have tests for cases with multiple attributes if they exist.

Reply:
The issue here is mistakenly temporarily treating HL intrinsics that have the same signature as the same intrinsic, which causes attributes as defined in gen_intrin_main to bleed into those that match. The only attributes there are `ro`, `rn`, and `wv`. There are none that have more than one.